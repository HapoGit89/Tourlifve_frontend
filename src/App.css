.App {
  text-align: center;
  font-family: myFont;
  width: 100%;
  height: 100%
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

body {
  background-image: url("https://images.unsplash.com/photo-1597158520886-43e23f1acb1e?q=80&w=1740&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D");
  background-size: cover;
  background-attachment: fixed;
}
