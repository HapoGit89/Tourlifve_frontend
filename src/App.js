
import './App.css';
import {React, useState, useEffect} from "react"
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import userContext from './userContext';
import MyNavBar from './Components/NavBar/MyNavBar';
import Home from './Components/Home/Home';
import SignUpForm from './Components/SignUpForm/SignUpForm'
import LoginForm from './Components/LoginForm/LoginForm';
import TourList from './Components/TourList/TourList';
import TourCreateForm from './Components/TourCreateForm/TourCreateForm';
import TourDetails from './Components/TourDetails/TourDetails';
import TourstopDetails from './Components/TourstopDetails/Tourstopdetails';
import UserForm from './Components/UserForm/UserForm';
import TourstopCreateForm from './Components/TourstopCreateForm/TourstopCreateForm';
import TourEditForm from './Components/TourEditForm/TourEditForm';
import TourDelete from './Components/TourDelete/TourDelete';
import { TourApi } from './api';
import 'bootstrap/dist/css/bootstrap.min.css';




function App() {
  const [user, setUser] = useState({})
 


  // catch userdata and set API Token when App mounts, cleanUp resets Api token
  useEffect(()=>{
    let username = localStorage.getItem("username")
    let token = localStorage.getItem("token")
    if (username && token){
      TourApi.token = token
      getUser(username)
    }
  
  }, [user.token])


  // gets userdata for given username and stores it in State
  const getUser = async(username)=>{
    const res = await TourApi.getUser(username)
    setUser({...res, token: TourApi.token})   
  }


  // User Login leads to localStorage Update, Api Token Update and State Update
  const logIn = (data)=>{
    localStorage.setItem("username", data.username)
    localStorage.setItem("token", data.token)
    TourApi.token = data.token
    getUser(data.username)
  }

  // clear localStorage when Logout and clear Api Token
  const logOut = ()=>{
    localStorage.clear()
     TourApi.token = ""
  }



  // // Apply current user to given JobId
  // const applyJob = async (jobId)=>{
  //   try{
  //   const res = await JoblyApi.ApplyForJob(user.username, jobId)
  //   if (res.applied){
  //     alert(`You just applied for Job ${res.applied}`)
  //     getUser(user.username)
  //   }
  //   }
  //   catch(e){
  //     alert(`Oops something went wrong!`)
  //   }
  // }


  return (
    <div className="App">
      <userContext.Provider value={user}>
   <BrowserRouter>
        <MyNavBar user={user} logOut={logOut}/>
        <Routes>

          <Route exact path="/signup" element={<SignUpForm logIn={logIn}/>}></Route>    
       
          <Route exact path="/login" element={<LoginForm logIn={logIn}/>}></Route>    
      
          <Route exact path="/tours" element={<TourList getUser={getUser}/>}></Route>    
      
          <Route exact path="/tours/new" element={<TourCreateForm/>}></Route>    
      
          <Route exact path="/tours/:id" element={<TourDetails/>}></Route>    
       
          <Route exact path="/tourstops/:id" element={<TourstopDetails/>}></Route>    
        
          <Route exact path="/users/:username" element={<UserForm getUser = {getUser}/>}></Route>    
       
          <Route exact path="/tours/:id/edit" element={<TourEditForm/>}></Route> 

           <Route exact path="/tours/:id/delete" element={<TourDelete/>}></Route>      

              <Route exact path="/tours/:tour_id/tourstops/new" element={<TourstopCreateForm/>}></Route>        
        
          <Route exact path="/" element={<Home></Home>}></Route>    
       
        </Routes>
        </BrowserRouter>
        </userContext.Provider>
      
    </div>
  );
}

export default App;
