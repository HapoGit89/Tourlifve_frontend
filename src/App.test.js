import { render, fireEvent } from '@testing-library/react';
import App from './App';

// Smoke Test
test('renders without crashing', () => {
  render(<App />);

});


// Snapshot Test
it("matches snapshot", function() {
  const {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});


// specialized render tests

it('renders Login Link ', ()=> {
  const {getByText} = render(<App/>)
  expect(getByText("Login")).toBeInTheDocument()
})


it('renders Login Form when clicking Login Link ', ()=> {
  const {getByText, getByLabelText} = render(<App/>)
 const button = getByText("Login")
 fireEvent.click(button)
 expect(getByLabelText("Password")).toBeInTheDocument()
 expect(getByText("Login:")).toBeInTheDocument()
})


it('renders Signup Form when clicking Signup Link ', ()=> {
  const {getByText, getByLabelText, debug} = render(<App/>)
 const button = getByText("Signup")
 fireEvent.click(button)
 expect(getByLabelText("Password:")).toBeInTheDocument()
 expect(getByText("Sign Up For Tourlifve:")).toBeInTheDocument()
})
