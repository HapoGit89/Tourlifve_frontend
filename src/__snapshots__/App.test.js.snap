// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <nav
        class="navbar navbar-expand-md"
      >
        <div
          class="container-fluid"
        >
          <a
            aria-current="page"
            class="NavbarBrand active"
            href="/"
          >
            Tourlifve ðŸšŒ
          </a>
          <ul
            class="ml-auto navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/login"
              >
                Login
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/signup"
              >
                Signup
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div
      class="WelcomeMessage"
    >
      <h1>
        Welcome to Tourlifve, please register or login!
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
